src/neural/factory.h

```diff
@@ -51,13 +52,37 @@ class NetworkFactory {
     Register(const std::string& name, FactoryFunc factory, int priority = 0);
   };
 
+  // Add the network/backend parameters to the options dictionary.
+  static void PopulateOptions(OptionsParser* options);
+
   // Returns list of backend names, sorted by priority (higher priority first).
   std::vector<std::string> GetBackendsList() const;
 
   // Creates a backend given name and config.
   std::unique_ptr<Network> Create(const std::string& network, const Weights&,
                                   const OptionsDict& options);
 
+  // Helper function to load the network from the options. Returns nullptr
+  // if no network options changed since the previous call.
+  static std::unique_ptr<Network> LoadNetwork(const OptionsDict& options);
+
+  // Parameter IDs.
+  static const OptionId kWeightsId;
+  static const OptionId kBackendId;
+  static const OptionId kBackendOptionsId;
+
+  struct BackendConfiguration {
+    BackendConfiguration() {}
+    BackendConfiguration(const OptionsDict& options);
+    std::string weights_path;
+    std::string backend;
+    std::string backend_options;
+    bool operator==(const BackendConfiguration& other);
```

Comment:
`const` (and operator!= also const)

Reply:
I missed this comment before merging. I got some compile error when I tried it earlier and gave up. I can make a follow-up if you think it is important.

Reply:
Probably not worth a separate PR, but I think I'll change that together with some other change, if not forget.

Reply:
I can do it with #357 later today.